cg/cli/deliver/base.py

```diff
@@ -130,3 +132,44 @@ def deliver_ticket(
     delivery_service.deliver_files_for_ticket(
         ticket_id=ticket, delivery_base_path=Path(inbox), dry_run=dry_run
     )
+
+
+@deliver.command(name="auto-fastq")
+@click.pass_obj
+@DRY_RUN
+def deliver_auto_fastq(context: CGConfig, dry_run: bool):
+    """
+    Deliver all case files based on delivery type to the customer inbox on the HPC for cases connected to a ticket.
+    1. get all cases with analysis type fastq that need to be delivered
+    2. check if their upload has started
+    3. if not, start the upload
+    4. update the uploaded at
+    5. commit the changes
+
+    """
+    """Starts upload of all not previously uploaded cases with analysis type fastq to
+       clinical-delivery."""
+    rsync_api: RsyncAPI = RsyncAPI(config=context)
+    service_builder = DeliveryServiceFactory(
+        store=context.status_db,
+        hk_api=context.housekeeper_api,
+        tb_service=context.trailblazer_api,
+        rsync_service=rsync_api,
+        analysis_service=context.analysis_service,
+    )
+
+    analyses: list[Analysis] = context.analysis_service.get_analyses_to_upload_for_workflow(
+        workflow=Workflow.FASTQ
+    )
+    for analysis in analyses:
+        case: Case = analysis.case
+        delivery_service: DeliverFilesService = service_builder.build_delivery_service(
+            delivery_type=DataDelivery.FASTQ,
+            workflow=case.data_analysis,
+        )
+        delivery_service.deliver_files_for_case(
+            case=case, delivery_base_path=Path(context.delivery_path), dry_run=dry_run
+        )
+        context.status_db.update_analysis_upload_started_at(
+            analysis_id=analysis.id, upload_started_at=datetime.now()
+        )
```

Comment:
Should we have try and catch here to not stop if one analysis fails?

Reply:
Yes, you are right!

Reply:
fixed